src/body.rs

```diff
@@ -0,0 +1,36 @@
+//! Tower <-> hyper body utilities
+
+use futures::Poll;
+use hyper::body::Payload;
+use tokio_buf::BufStream;
+
+/// Lifts a body to support `Payload` and `BufStream`
+#[derive(Debug)]
+pub struct LiftBody<T> {
+    inner: T,
+}
+
+impl<T> LiftBody<T> {
+    /// Lifts the inner `T`
+    pub fn new(inner: T) -> Self {
+        LiftBody { inner }
+    }
+}
+
+impl<T: Payload> BufStream for LiftBody<T> {
+    type Item = <T as Payload>::Data;
```

Comment:
Can you not just do T::Data here?

Reply:
Yes, but I prefer to be explicit from which traits associated type im pulling from.